File: Listas Dikastis/Lista 1/G-binario_para_inteiro.hs
btoi :: String -> Int
btoi binario = foldl (\soma digito -> soma * 2 + (charParaInt digito)) 0 binario
  where
    charParaInt :: Char -> Int
    charParaInt '0' = 0
    charParaInt '1' = 1

main = do
    s <- getLine
    let result = btoi s
    print result

File: Listas Dikastis/Lista 1/G-binario_para_inteiro.py
def btoi(binario: str) -> int:
    soma = 0
    tam = len(binario)
    for i in range(tam):
        soma += (int(binario[i]) * (2**(tam-1-i)))
    return soma

def btoi2(binario: str) -> int:
    if (len(binario) == 1):
        return int(binario)
    
    ultimo = binario[-1]
    resto = binario[:-1]
    
    return btoi(resto) * 2 + int(ultimo)
    
print(btoi2("1010"))